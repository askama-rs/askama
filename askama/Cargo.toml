[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
