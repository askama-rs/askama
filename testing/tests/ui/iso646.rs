use askama::Template;

#[derive(Template)]
#[template(ext = "txt", source = "{{ a & b }}")]
struct BitAnd {
    a: u32,
    b: u32,
}

#[derive(Template)]
#[template(ext = "txt", source = "{{ a bitand b }}")]
struct BitAndIso646 {
    a: u32,
    b: u32,
}

#[derive(Template)]
#[template(ext = "txt", source = "{{ a bitor b }}")]
struct BitOrIso646 {
    a: u32,
    b: u32,
}

#[derive(Template)]
#[template(ext = "txt", source = "{{ a ^ b }}")]
struct Xor {
    a: u32,
    b: u32,
}

#[derive(Template)]
#[template(ext = "txt", source = "{{ a xor b }} {{ c }}")]
struct XorIso646 {
    a: u32,
    b: u32,
}

fn main() {
}
