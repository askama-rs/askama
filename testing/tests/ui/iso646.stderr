error: the binary AND operator is called `bitand` in askama
 --> <source attribute>:1:6
       " b }}"
 --> tests/ui/iso646.rs:4:34
  |
4 | #[template(ext = "txt", source = "{{ a & b }}")]
  |                                  ^^^^^^^^^^^^^

error: the binary XOR operator is called `xor` in askama
 --> <source attribute>:1:6
       " b }}"
  --> tests/ui/iso646.rs:25:34
   |
25 | #[template(ext = "txt", source = "{{ a ^ b }}")]
   |                                  ^^^^^^^^^^^^^

error[E0609]: no field `c` on type `&XorIso646`
  --> tests/ui/iso646.rs:31:10
   |
31 | #[derive(Template)]
   |          ^^^^^^^^ unknown field
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
